index.js

```diff
@@ -4302,10 +4303,11 @@
   //. > S.stripSuffix ('.md') ('README')
   //. Nothing
   //. ```
-  const stripSuffix = suffix => s => {
-    const idx = s.length - suffix.length;  // value may be negative
-    return s.slice (idx) === suffix ? Just (s.slice (0, idx)) : Nothing;
-  };
+  const stripSuffix = suffix => s => (
+    s.endsWith (suffix) ?
+    Just (s.slice (0, s.length - suffix.length)) :
```

Comment:
It looks as though we could use `s.slice (0, -suffix.length)` here. If `suffix` is `''`, though, that expression would evaluate to the empty string rather than to (the entirety of) the string being sliced. We have a test for this. :)